cmd/gctsDeployCommit.go

```diff
@@ -0,0 +1,82 @@
+package cmd
+
+import (
+	"io/ioutil"
+	"net/http/cookiejar"
+
+	gabs "github.com/Jeffail/gabs/v2"
+	"github.com/SAP/jenkins-library/pkg/command"
+	piperhttp "github.com/SAP/jenkins-library/pkg/http"
+	"github.com/SAP/jenkins-library/pkg/log"
+	"github.com/SAP/jenkins-library/pkg/telemetry"
+	"github.com/pkg/errors"
+)
+
+func gctsDeployCommit(config gctsDeployCommitOptions, telemetryData *telemetry.CustomData) {
+	// for command execution use Command
+	c := command.Command{}
+	// reroute command output to logging framework
+	c.Stdout(log.Entry().Writer())
+	c.Stderr(log.Entry().Writer())
+
+	// for http calls import  piperhttp "github.com/SAP/jenkins-library/pkg/http"
+	// and use a  &piperhttp.Client{} in a custom system
+	// Example: step checkmarxExecuteScan.go
+	httpClient := &piperhttp.Client{}
+
+	// error situations should stop execution through log.Entry().Fatal() call which leads to an os.Exit(1) in the end
+	err := deployCommit(&config, telemetryData, &c, httpClient)
+	if err != nil {
+		log.Entry().WithError(err).Fatal("step execution failed")
+	}
+}
+
+func deployCommit(config *gctsDeployCommitOptions, telemetryData *telemetry.CustomData, command execRunner, httpClient piperhttp.Sender) error {
+
+	cookieJar, cookieErr := cookiejar.New(nil)
+	if cookieErr != nil {
+		return errors.Wrap(cookieErr, "deploy commit failed")
+	}
+	clientOptions := piperhttp.ClientOptions{
+		CookieJar: cookieJar,
+		Username:  config.Username,
+		Password:  config.Password,
+	}
+	httpClient.SetOptions(clientOptions)
+
+	url := config.Host +
+		"/sap/bc/cts_abapvcs/repository/" + config.Repository +
+		"/pullByCommit?sap-client=" + config.Client
+	if config.Commit != "" {
+		url = url + "&request=" + config.Commit
+	}
+
+	resp, httpErr := httpClient.SendRequest("GET", url, nil, nil, nil)
+
+	defer func() {
+		if resp != nil && resp.Body != nil {
+			resp.Body.Close()
+		}
+	}()
+
+	if resp == nil || httpErr != nil {
+		return errors.Errorf("deploy commit failed: %v", httpErr)
+	}
+
+	bodyText, readErr := ioutil.ReadAll(resp.Body)
+
+	if readErr != nil {
+		return errors.Wrapf(readErr, "deploying commit failed")
+	}
+
+	response, parsingErr := gabs.ParseJSON([]byte(bodyText))
+
+	if parsingErr != nil {
+		return errors.Wrap(parsingErr, "deploying commit failed")
```

Comment:
Why not putting the bodyText into the error message? Might help to understand the details when troubleshooting. E.g. maybe there is a http-4xx and we get some plain text in the response body outlining e.g. what was wrong with the request. That cannot be parsed, but might nevertheless be helpful for troubleshooting. Only valid case where we have to omit the response body would be when we expect some secrets inside.

Reply:
Since this is dynamic parsing, I expect no parsing problem for a http-4xx response. But it might be a good idea to return the body in any case.

Reply:
> I expect no parsing problem for

I did not check the details wrt gabs parsing, but I would expect parsing issues in case we have not well formed json in the response body. E.g. plain text response.

Now you might argue that "your server" returns _always_ some valid json, also in cases of 4xx and 5xx, since you might now some implemention details. In that case I would  in turn respond:

 1.) not sure how you can guarantee that for each and every 5xx case.
2.) it might happen that a response is issued by some proxies, nginx load balancers, etc and not by "your server".